crates/bevy_tasks/src/slice.rs

```diff
@@ -1,6 +1,35 @@
 use super::TaskPool;
 
+/// Provides functions for mapping read-only slices across a provided [`TaskPool`].
 pub trait ParallelSlice<T: Sync>: AsRef<[T]> {
+    /// Splits the slice in chunks of size `chunks_size` or less and maps the chunks
+    /// in parallel across the provided `task_pool`. One task is spawned in the task pool
+    /// for every chunk.
+    ///
+    /// Returns a `Vec` of the mapped results.
```

Comment:
Could be nice to mention whether the order of items in the resulting vec is guaranteed to match the one of the slice. Same for the methods below.